src/server/info_collector.cpp

```diff
@@ -316,7 +316,8 @@ info_collector::get_hotspot_calculator(const std::string &app_name, const int pa
     if (iter != _hotspot_calculator_store.end()) {
         return iter->second;
     }
-    auto calculator = std::make_shared<hotspot_partition_calculator>(app_name, partition_count);
+    auto calculator =
+        std::make_shared<hotspot_partition_calculator>(app_name, partition_count, _shell_context);
```

Comment:
I think ddl_client is necessary for `hotspot_partition_calculator`, you can create a new ddl client instance for calculator, not share the shell_context with info_collector.

Reply:
There will be a lot of `hotspot_partition_calculator`(s) in one database, I think it is better to share one instance